src/main/java/seedu/siasa/model/policy/UniquePolicyList.java

```diff
@@ -74,7 +74,8 @@ public void remove(Policy toRemove) {
      */
     public void removeBelongingTo(Person owner) {
         requireNonNull(owner);
-        for (Policy policy: internalList) {
+        for (int i = internalList.size() - 1; i >= 0; i--) {
+            Policy policy = internalList.get(i);
             if (policy.getOwner().equals(owner)) {
                 remove(policy);
             }
```

Comment:
This works! Could we consider using a built-in function removeIf instead to make this function even cleaner? So somthing like this. `internalList.removeIf(policy -> policy.getOwner().equals(owner));`